File: Data Structures & Algorithms/climbing-stairs/submission-3.cpp
class Solution {
public:
// int func(int n,vector<int> &dp){
//     if(n<=1) return 1;
//     if(dp[n]!=-1) return dp[n];
//     return dp[n]=func(n-2,dp)+func(n-1,dp);
// }
    int climbStairs(int n) {
        vector<int> dp(n+1,-1);
        dp[0]=1;
        dp[1]=1;
        for(int i=2;i<=n;i++){
            dp[i]=dp[i-2]+dp[i-1];
        }
       return dp[n];
        
    }
};

File: Data Structures & Algorithms/minimum-window-with-characters/submission-3.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        n=len(s)
        k=len(t)
        count=defaultdict(int)
        for i in range(k):
            count[t[i]]+=1
        need=len(count)
        have=0
        l=0
        minlen=math.inf
        res=[-1,-1]
        count2=defaultdict(int)
        for r in range(n):
            count2[s[r]]+=1
            if count[s[r]]==count2[s[r]]: have+=1
            while(have==need):
                if r-l+1 < minlen:
                    res=[l,r]
                    minlen=r-l+1
               
                count2[s[l]]-=1
                if count2[s[l]]<count[s[l]] : have-=1
                l+=1
        return s[res[0]:res[1]+1]

        
        